test/controllers/VatCertificateControllerSpec.scala

```diff
@@ -41,15 +43,19 @@ class VatCertificateControllerSpec extends ControllerBaseSpec {
     val mockEnrolmentsAuthService: EnrolmentsAuthService = new EnrolmentsAuthService(mockAuthConnector)
     val mockAccountDetailsService: AccountDetailsService = mock[AccountDetailsService]
     val mockServiceInfoService: ServiceInfoService = mock[ServiceInfoService]
-    val mockHybridUserPredicate: HybridUserPredicate = new HybridUserPredicate(mockAccountDetailsService, mockServiceErrorHandler)
-    val mockAgentPredicate: AgentPredicate = new AgentPredicate(mockEnrolmentsAuthService, messages, mockAppConfig)
+    val mockHybridUserPredicate: HybridUserPredicate = new HybridUserPredicate(mockAccountDetailsService, mockServiceErrorHandler, mcc, ec)
```

Comment:
There's a bunch of stuff being created in this trait that already exists in `ControllerBaseSpec`. Can you re-use those vals where possible?